src/utils/perfherder/chartJs/perfherderFormatter.js

```diff
@@ -17,28 +17,29 @@ const generateInitialOptions = series => {
     tooltips: {
       callbacks: {
         footer: (tooltipItems, data) => {
-          const tooltipData = [];
-          let delta = 'n/a';
-          let deltaPercentage = 'n/a';
-          let dataset = 'n/a';
-          let currentData = 'n/a';
+          // get data from all points of selected series
+          const dataset = data.datasets[tooltipItems[0].datasetIndex].data;
+          // get data from selected point
+          const currentData = dataset[tooltipItems[0].index].y;
+          const tooltipData = []; // footer's text lines will be stored here
+          let delta = 0;
+          let deltaPercentage = 0;
 
           if (tooltipItems[0].index > 0) {
-            dataset = data.datasets[tooltipItems[0].datasetIndex].data;
-
-            currentData = dataset[tooltipItems[0].index].y;
             const previousData = dataset[tooltipItems[0].index - 1].y;
 
-            delta = (currentData - previousData).toFixed(2);
-            deltaPercentage = (
-              ((currentData - previousData) / previousData) *
-              100
-            ).toFixed(2);
+            delta = currentData - previousData;
+            // [(c - p) / p] * 100 is equivalent to (c / p - 1) * 100
```

Comment:
These are good comments: they explain the next line differently, helping the reader understand what it is doing.

Reply:
Thank you

Reply:
I didn't mean to contradict @aimenbatool - on balance, I think it's best to keep comments except when they are totally redundant.  I often see things like:
```js
// set x to 0
x = 0;
```
or
```js
// export DataComponent
module.exports = DataComponent;
```
and those comments are definitely worth deleting because they don't help the reader.  But your comments here help someone not familiar with the code to figure out what it means a little faster, so I think they can stay.

Reply:
@2alin Keep the comments :) I go with what Dustin says.

Reply:
Thank you @djmitche and @aimenbatool, these little discussions are good for me because I end up understanding how you all think as team :smiley:

Reply:
I agree it is helpful to have the comments in this situation. Some of the best advice I've been given is to name variables and functions in a way that eliminates the need for a comment where possible.
So instead of:
```
// get data from all points of selected series
const dataset = data.datasets[tooltipItems[0].datasetIndex].data;
```

You could write:
`const allSeriesData` or `const selectedSeriesData`

Don't feel you need to change it - this is something to consider for future work. :)